presto-main/src/main/java/com/facebook/presto/execution/QueryInfo.java

```diff
@@ -60,6 +60,8 @@
     private final String updateType;
     private final Optional<StageInfo> outputStage;
     private final FailureInfo failureInfo;
+    private final Optional<String> failureTask;
```

Comment:
I remember our discussion now - we can still pull this stuff into another wrapper class that has the stuff QueryInfo needs - maybe QueryFailureInfo?